spec-tests/src/test/java/org/candlepin/spec/EntitlementCertificateV3SpecTest.java

```diff
@@ -17,121 +17,493 @@
+import org.candlepin.dto.api.v1.DistributorVersionCapabilityDTO;
+import org.candlepin.dto.api.v1.DistributorVersionDTO;
+import org.candlepin.dto.api.v1.EntitlementDTO;
 import org.candlepin.dto.api.v1.OwnerDTO;
 import org.candlepin.dto.api.v1.PoolDTO;
 import org.candlepin.dto.api.v1.ProductDTO;
+import org.candlepin.dto.api.v1.ProvidedProductDTO;
+import org.candlepin.dto.api.v1.SubscriptionDTO;
+import org.candlepin.resource.HostedTestApi;
 import org.candlepin.resource.OwnerApi;
 import org.candlepin.resource.OwnerContentApi;
 import org.candlepin.resource.OwnerProductApi;
+import org.candlepin.spec.bootstrap.assertions.CandlepinMode;
 import org.candlepin.spec.bootstrap.client.ApiClient;
 import org.candlepin.spec.bootstrap.client.ApiClients;
 import org.candlepin.spec.bootstrap.client.SpecTest;
 import org.candlepin.spec.bootstrap.client.api.ConsumerClient;
+import org.candlepin.spec.bootstrap.data.builder.Branding;
+import org.candlepin.spec.bootstrap.data.builder.ConsumerTypes;
 import org.candlepin.spec.bootstrap.data.builder.Consumers;
 import org.candlepin.spec.bootstrap.data.builder.Contents;
 import org.candlepin.spec.bootstrap.data.builder.Owners;
 import org.candlepin.spec.bootstrap.data.builder.Pools;
 import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.builder.Subscriptions;
 import org.candlepin.spec.bootstrap.data.util.CertificateUtil;
 import org.candlepin.spec.bootstrap.data.util.StringUtil;
 import org.candlepin.spec.bootstrap.data.util.X509HuffmanDecodeUtil;
 
 import com.fasterxml.jackson.databind.JsonNode;
 
 import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
 
 
 @SpecTest
 public class EntitlementCertificateV3SpecTest {
 
     private static ApiClient client;
     private static OwnerDTO owner;
+    private static ConsumerDTO system;
+    private static ProductDTO product;
+    private static ProductDTO product30;
+    private static ContentDTO content;
+    private static ContentDTO archContent;
+    private static PoolDTO pool;
+    private static PoolDTO pool30;
+
+    private static SubscriptionDTO subscription;
+    private static SubscriptionDTO subscription30;
     private OwnerApi ownerApi;
     private OwnerContentApi ownerContentApi;
     private OwnerProductApi ownerProductApi;
     private ConsumerClient consumerApi;
+    private HostedTestApi hostedTestApi;
 
     @BeforeAll
     public void beforeAll() throws ApiException {
         client = ApiClients.admin();
         ownerApi = client.owners();
-        owner = ownerApi.createOwner(Owners.random());
         ownerContentApi = client.ownerContent();
         ownerProductApi = client.ownerProducts();
         consumerApi = client.consumers();
+        hostedTestApi = client.hosted();
```

Comment:
Can we move all these clients in the BeforeEach, and delete the BeforeAll, to minimize the state sharing between tests?

Reply:
Do we care about the tests sharing a client? It should be immutable and thread safe.

Reply:
Now that they are in the BeforeEach, there should not be a sharing issue